codalab/worker_manager/gcp_batch_worker_manager.py

```diff
@@ -0,0 +1,114 @@
+import json
+import logging
+import os
+import uuid
+import yaml
+from argparse import ArgumentParser
+from typing import Any, Dict, List
+
+from .worker_manager import WorkerManager, WorkerJob
+
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class GCPBatchWorkerManager(WorkerManager):
+    NAME: str = 'gcp-batch'
+    DESCRIPTION: str = 'Worker manager for submitting jobs to Google Cloud Platform via Kubernetes'
+
+    @staticmethod
+    def add_arguments_to_subparser(subparser: ArgumentParser) -> None:
+        # GCP arguments
+        subparser.add_argument('--project', type=str, help='GCP project', required=True)
+        subparser.add_argument('--bucket', type=str, help='GCP bucket for logs', required=True)
+
+        # Job-related arguments
+        subparser.add_argument(
+            '--cpus', type=int, default=1, help='Default number of CPUs for each worker'
+        )
+        subparser.add_argument(
+            '--gpus', type=int, default=0, help='Default number of GPUs to request for each worker'
+        )
+        subparser.add_argument(
+            '--memory-mb', type=int, default=2048, help='Default memory (in MB) for each worker'
+        )
+
+        # Output path
+        subparser.add_argument(
+            '--output-path', type=str, default='.', help='Path to the output directory'
+        )
+
+    def __init__(self, args):
+        super().__init__(args)
+
+        self.codalab_username = os.environ.get('CODALAB_USERNAME')
+        self.codalab_password = os.environ.get('CODALAB_PASSWORD')
+        if not self.codalab_username or not self.codalab_password:
+            raise EnvironmentError(
+                'Valid credentials need to be set as environment variables: CODALAB_USERNAME and CODALAB_PASSWORD'
+            )
+
+    def get_worker_jobs(self) -> List[WorkerJob]:
+        # Use kubectl to get the current running pods in JSON format
+        output_json: str = self.run_command(
+            ['kubectl', 'get', 'pod', '--field-selector=status.phase==Running', '--output=json']
```

Comment:
Should we use the Python Kubernetes client (https://github.com/kubernetes-client/python) instead? This way, the pip package will handle the dependency instead of requiring people to install `kubectl` on the worker manager machine beforehand.

Reply:
+1 for this - this is probably the best approach

Reply:
@teetone I think all the documentation is available on this file (https://raw.githubusercontent.com/kubernetes-client/python/master/kubernetes/docs/CoreV1Api.md).

For example, this particular command corresponds to the `list_pod_for_all_namespaces` function:

![image](https://user-images.githubusercontent.com/1689183/111841505-71493700-88d4-11eb-849a-60d7aab8be35.png)

Reply:
@epicfaace Thanks for finding this. I will take a look.